lib/phoenix/live_dashboard/system_info.ex

```diff
@@ -218,10 +228,23 @@ defmodule Phoenix.LiveDashboard.SystemInfo do
 
     next_state = for {_sorter, info} <- processes, into: %{}, do: {info[:pid], info[:reductions]}
 
-    count = if search, do: length(processes), else: :erlang.system_info(:process_count)
+    count = length(processes)
     processes = processes |> Enum.sort() |> Enum.take(limit) |> Enum.map(&elem(&1, 1))
 
-    {processes, count, next_state}
+    {active_filter, available_filters, processes, count, next_state}
+  end
+
+  def get_process_filter_data(filter) do
+    case Application.get_env(:phoenix_live_dashboard, :process_filter) do
+      nil ->
+        {nil, nil, Process.list()}
+
+      filter_mod ->
+        available_filters = filter_mod.list()
+        active_filter = (filter in available_filters && filter) || filter_mod.default_filter()
```

Comment:
Not sure how to handle the situation of `filter == nil`. I think we can always add the "default filter" that doesn't filter any process at all when `filter == nil` . If we handle it here, the user doesn't need to implement it always in the `filter_mod`. So when `filter == nil` we can return `Process.list()`. However, it'd also require extra changes in some places so not sure if it's worth it.

Reply:
The base implementation of `default_filter` is provided here:
https://github.com/phoenixframework/phoenix_live_dashboard/blob/a74954b6cc230d9ccf366665de5138d887134208/lib/phoenix/live_dashboard/page_filter.ex#L62-L71

, so the user doesn't need to implement it. So `filter == nil` should not cause any issues I think. At least the issue doesn't show up in manual testing :-)